libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -850,6 +850,116 @@ bool AP_Arming::camera_checks(bool display_failure)
     return true;
 }
 
+// request an auxiliary authorisation id.  This id should be used in subsequent calls to set_aux_auth_passed/failed
+// returns true on success
+bool AP_Arming::get_aux_auth_id(uint8_t& auth_id)
+{
+    WITH_SEMAPHORE(aux_auth_sem);
+
+    // check we have enough room to allocate another id
+    if (aux_auth_count >= aux_auth_count_max) {
+        aux_auth_error = true;
+        return false;
+    }
+
+    // allocate buffer for failure message
+    if (aux_auth_fail_msg == nullptr) {
+        aux_auth_fail_msg = (char *)calloc(aux_auth_str_len, sizeof(char));
+        if (aux_auth_fail_msg == nullptr) {
+            aux_auth_error = true;
+            return false;
+        }
+    }
+    auth_id = aux_auth_count;
+    aux_auth_state[auth_id] = AuxAuthStates::AUX_AUTH_STATE_NO_RESPONSE;
```

Comment:
This is unneeded once this state is removed.

Reply:
ok, removed